File: backend/components/metiers/threed/tests_centralises/fixtures/samples/test_samples.py
# Test avancé pour samples.py du module fixtures/samples
import importlib.util
import sys
import os
import pytest

sample_fixture_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../../fixtures/samples/sample_fixture.py'))
# Correction définitive du chemin pour pointer vers le bon fichier fixture dans la structure threed.
spec = importlib.util.spec_from_file_location('sample_fixture', sample_fixture_path)
sample_fixture = importlib.util.module_from_spec(spec)
sys.modules['sample_fixture'] = sample_fixture
spec.loader.exec_module(sample_fixture)
sample_helper_fixture = sample_fixture.sample_helper_fixture

# On n'a plus besoin de redéfinir la fixture ici, pytest la gère automatiquement

def test_fixtures_samples():
    assert True

def test_sample_helper_fixture_nominal(sample_helper_fixture):
    val = sample_helper_fixture
    assert val["name"] == "Sample Helper"
    assert val["type"] == "helper"
    assert val["status"] == "active"

def test_sample_helper_fixture_edge_case(sample_helper_fixture):
    val = sample_helper_fixture
    assert isinstance(val, dict)
    assert set(val.keys()) == {"name", "type", "status"}

File: backend/components/metiers/threed/tests_centralises/fixtures/samples/test_samples.js
// test_samples.js – Tests ultra avancés pour fixtures/samples (JS)
const { sample } = require('../../../fixtures/samples/sample_fixture');

describe('Fixtures Samples', () => {
  it('should expose all required samples', () => {
    expect(sample).toBeDefined();
    expect(sample.name).toBe('Sample Helper');
    expect(sample.type).toBe('helper');
    expect(sample.status).toBe('active');
  });

  it('should respect business logic for samples', () => {
    expect(typeof sample).toBe('object');
    expect(Object.keys(sample)).toEqual(['name', 'type', 'status']);
  });

  it('should handle edge cases (immutability)', () => {
    expect(() => { sample.name = 'Changed'; }).not.toThrow();
  });
});
